PORTAL/jobs/_common.py

```diff
@@ -166,44 +207,46 @@ def __fspath__(self):
         return str(self.request)
 
     @property
-    def requestsroot(self):
+    def requestsroot(self) -> str:
         try:
             return self.request.requestsroot
         except AttributeError:
             #return self.jobs.requests.root
             return os.path.dirname(self.request.root)
 
     @property
-    def resultsroot(self):
+    def resultsroot(self) -> str:
         try:
             return self.result.resultsroot
         except AttributeError:
             #return self.jobs.results.root
             return os.path.dirname(self.result.root)
 
     @property
-    def job_script(self):
+    def job_script(self) -> str:
         return self.work.job_script
 
     @property
-    def pidfile(self):
+    def pidfile(self) -> str:
         return self.work.pidfile
 
     @property
-    def logfile(self):
+    def logfile(self) -> str:
         return self.work.logfile
 
-    def look_up(self, name, subname=None):
+    def look_up(self, name: str, subname: Optional[str] = None) -> Any:
         value = getattr(self, name)
         if subname:
             value = getattr(value, subname)
         return value
 
-    def copy(self):
+    def copy(self) -> "JobFS":
+        # XXX This may not work, since it doesn't do a deep copy of the first
+        # three arguments
```

Comment:
This comment isn't needed any more, right?

Reply:
Yeah, good catch.  With the recent changes, this should be fine.